pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -211,24 +216,28 @@ def on_run_end(self) -> None:
 
         # get the model and call model.training_epoch_end
         model = self.trainer.lightning_module
-        if is_overridden("training_epoch_end", model) and self._epoch_output:
-            processed_outputs = self._prepare_outputs(self._epoch_output, batch_mode=False)
+        if is_overridden("training_epoch_end", model) and self._outputs:
+            epoch_end_outputs = self._prepare_outputs_training_epoch_end(
+                self._outputs,
+                automatic=model.automatic_optimization,
+                num_optimizers=len(self.trainer.optimizers),
+            )
             # check that the dataloader/iterator produced a batch
-            if processed_outputs:
+            # FIXME: still necessary?
+            if epoch_end_outputs:
```

Comment:
@carmocca should we make a separate PR for this? otherwise the diff here gets destroyed :)

Reply:
For what exactly? The variable rename? The diff looks okay to me

Reply:
the fixme, which I interpret as "is this if statement still necessary".

Reply:
ah sure, we can check that in a separate PR

Reply:
```suggestion
            if epoch_end_outputs:
```